dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -299,133 +339,167 @@ class _OperatorManager:
     Uses :class:`ops.Operator` to create OpSpec and handle input sets.
     """
 
-    def __init__(self, op_class, seed, inputs, kwargs):
+    def __init__(self, op_class, op_name, next_logical_id, batch_size, seed, inputs, kwargs):
         """Creates direct operator."""
 
         self._separate_kwargs(kwargs)
 
-        if op_class.__name__ == 'ArithmeticGenericOp':
+        if op_name == 'arithmetic_generic_op':
             inputs = self._init_arithm_op(kwargs['name'], inputs)
 
         # Save inputs classification for later verification.
-        self.inputs_classification = [_Classification(input) for input in inputs]
+        self._inputs_classification = []
+
+        # When using input sets we have to create separate operators for each input.
+        input_set_len = 1
+        for i, input in enumerate(inputs):
+            classification = _Classification(input, f'Input {i}')
+
+            if isinstance(classification.is_batch, list):
+                if input_set_len == 1:
```

Comment:
Wouldn't a -1 or something make more sense for the initial value?

Reply:
done